codegen/build-clients.sh

```diff
@@ -0,0 +1,70 @@
+#!/bin/bash
```

Comment:
I was modeling this after the work @jhamon had done in other repos, but this diverges a bit in that I'm not using a docker image, and we're doing both OAS and proto files in here.

Let me know if this doesn't make sense - or any other feedback. I didn't want to make too many changes to the dependencies we were using to generate code.